### PR TITLE
Use SumoClient for uploader

### PR DESCRIPTION
**Pull request by thezultimate**
In this PR, the uploader is shifted from using `CallSumoApi` to `SumoClient`.

### CODE REVIEW

File: src/fmu/sumo/uploader/_caseondisk.py [hunk 1]
```diff
@@ -129,7 +129,8 @@ def _get_sumo_parent_id(self):
         Call sumo, check if the case is already there. Use fmu_case_uuid for this."""
 
         query = f"fmu.case.uuid:{self.fmu_case_uuid}"
-        search_results = self.sumo_connection.api.searchroot(query, search_size=2)
+        path = f"/searchroot?$query={query}&$from={0}&$size={2}"
+        search_results = self.sumo_connection.api.get(path)
```
Comment (adnejacobsen) [inline src/fmu/sumo/uploader/_caseondisk.py:133]: In `SumoClient.get` you dont have to put the query parameters in the path. `kwargs` will be treated like query parameters so you can do stuff like this:

```python
sumo.get("/searchroot",  query=f"fmu.case.uuid:{self.fmu_case_uuid}", size=2)
```

File: src/fmu/sumo/uploader/_connection.py [hunk 2]
```diff
@@ -43,4 +34,4 @@ def refresh(self):
 
     def _establish_connection(self):
         """Establish the connection with Sumo API, take user through 2FA."""
-        return CallSumoApi(env=self.env)
+        return SumoClient(env=self.env)
```
Comment (adnejacobsen) [inline src/fmu/sumo/uploader/_connection.py:37]: `SumoClient` won't prompt for user login until a request has been made and we detect no valid token in the cache, so technically the `_establish_connection` won't establish the connection. Not sure if it matters.

File: src/fmu/sumo/uploader/_caseondisk.py (hunk 1 above)
Comment (thezultimate) [reply src/fmu/sumo/uploader/_caseondisk.py]: This is applied in the latest commit. Please check :)

File: src/fmu/sumo/uploader/_connection.py (hunk 2 above)
Comment (thezultimate) [reply src/fmu/sumo/uploader/_connection.py:37]: The logic was left unchanged in this PR. Maybe we can keep it as-is until next iteration (if necessary).

**Review by adnejacobsen — APPROVED** [review]
LGTM 😃

**Pull request merged**